File: src/main/kotlin/community/flock/localpipeline/model/WorkItem.kt
package community.flock.localpipeline.model

data class WorkItem(
    val id: String,
    val title: String,
    val description: String? = null,
    val status: String? = null,
    val priority: String? = null,
    val labels: List<String> = emptyList(),
    val source: String,
    val team: String? = null,
    val url: String? = null,
)

File: src/model/work_item.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorkItem {
    pub id: String,
    /// Original ID from the source system, used for API calls (e.g. full Trello card ID, Linear UUID)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub source_id: Option<String>,
    pub title: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub status: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub priority: Option<String>,
    #[serde(default)]
    pub labels: Vec<String>,
    pub source: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub team: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub url: Option<String>,
}
